### DISCUSSION TITLE
Welcome to cordova-background-geolocation-plugin Discussions!

### DISCUSSION THREAD

**Discussion by RaddishIoW** [HaylLtd/cordova-plugin-background-geolocation#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by HarelM** [comment]
Hi,

My name is Harel and I'm the co-founder of [Israel Hiking Map](https://israelhiking.osm.org.il/).
I've been maintaining this open source project since 2013 and in 2018 we started using Cordova to allow maintaining a site and app in iOS and android.
The app is a navigation app and so the ability to record a route when the phone screen is off is a very important feature.
Which brought me here in order to keep this core functionality related plugin alive and kicking :-)

I hope other will join as well :-)

**Comment by RaddishIoW** [comment]
Hi,

My name is Adam and I'm the founder of Hayl Ltd, and we make a taxi management system. I use this plugin to monitor and track drivers as they move around, to give the taxi dispatchers an overview of where they are at any given time.
I've been developing for Mac, iOS, web and lately React/Cordova for about 20 years.

This plugin is essential to my software, and I'd like to make sure that others benefit from it as I do - I hope others will add their support to help keep this plugin relevant and useful.

**Comment by zdevaty** [comment]
Hi, Zdenek here. This plugin is also the key part of my fleet management app. A problem with newer Android versions led me to move from the now-unsupported Mauron85's fork here. I am glad you guys decided to take care of this plugin.

Thanks to your fork's docs, I found that my problem roots in the WebView callbacks that are not responding while the app is in the background. So I am wondering how to get around it. Is using the plugin's POST inevitable?

I have tried to use getLocations() and successfully got all the locations, even those missed by the callbacks. But these all have the same Location.time, so I can't order them and make a continuous path.

Do you have any tips for me? Or should I just use the plugin's POST functionality?
Thanks.

**Reply by HarelM** [reply]
See #19 and #29.
Feel free to submit a PR if you think the docs should be better. I use getLocation and haven't seen issues with the timestamp... Every helping hand would be appreciated :-)

**Reply by RaddishIoW** [reply]
Hi @MrMid, glad you've found this plugin useful!

I had exactly the same issue - on iOS the callbacks were being called even when the app was in the background, but on Android it wasn't consistently calling them. In the end I used the plugin's POST function instead of handling it in the callback app-side. This seems to be a limitation on the Android platform - I know we've discussed this before, and HarelM spoke with a cordova dev team member about this.

It's interesting that you're seeing that getLocations() is returning all the locations but with the same time... I wonder if this should be raised as a bug? I've not looked, but it makes sense to me that they'd have the times that they were recorded. @HarelM what do you think? I don't actually use that part of the API myself...

**Reply by RaddishIoW** [reply]
Ah! Our messages crossed in the ether Harel! :)

**Reply by zdevaty** [reply]
Thanks for your replies! I will try to look deeper into it later and possibly create an issue. I would like to avoid using the plugin's network functionality.

**Reply by zdevaty** [reply]
@RaddishIoW do you use the getLocations() method? Does it work well for you?

**Comment by pmolinal** [comment]
Hello people, first of all thanks for keeping this plugin alive. I want to use the mauron85 base but I have found the problems that you have already addressed. My question is how can I install this fork if I already have the original one working? Do you have any guide or steps that I do not want to break with everything?

Ionic:

   Ionic CLI                     : 6.10.1 (C:\Users\patricio\AppData\Roaming\npm\node_modules\@ionic\cli)
   Ionic Framework               : @ionic/angular 5.0.4
   @angular-devkit/build-angular : 0.803.25
   @angular-devkit/schematics    : 8.3.25
   @angular/cli                  : 8.3.25
   @ionic/angular-toolkit        : 2.2.0

Cordova:

   Cordova CLI       : 10.0.0
   Cordova Platforms : android 8.1.0
   Cordova Plugins   : cordova-plugin-ionic-keyboard 2.2.0, cordova-plugin-ionic-webview 4.1.3, (and 12 other plugins)

Utility:

   cordova-res : 0.11.0
   native-run  : 1.2.2

System:

   Android SDK Tools : 26.1.1 (C:\Users\patricio\AppData\Local\Android\Sdk)
   NodeJS            : v12.16.1 (C:\Program Files\nodejs\node.exe)
   npm               : 6.13.4
   OS                : Windows 10

thank you and sorry for my english as a translator

**Reply by HarelM** [reply]
Like any plugin, you need to uninstall the other one and install this one... They have different name in npm so they don't "collide".

**Comment by pmolinal** [comment]
Yes I understand, but what gives me doubt is that I have to uninstall 
the cordova plugin and npm and only install your npm?

according to this 
https://ionicframework.com/docs/native/background-geolocation or should 
I just replace what is npm?

**Reply by HarelM** [reply]
You need to uninstall the old one and install the new one using cordova cli:
`crodova plugin remove old...`
`cordova plugin add this-one`
Or any other way you see fit.

**Reply by pmolinal** [reply]
Hi guys, I haven't been able to get the plugin to work... I'm working on ionic... I've uninstalled the original but I manage to compile the app but it keeps asking me for the other plugin.

I have tried installing it as a cordova plugin and by npm... the other plugin has to be installed as well?

**Reply by HarelM** [reply]
This plugin is self contained. It doesn't need other plugins.

**Reply by RaddishIoW** [reply]
You might need to check through the Cordova config.xml for references to the old plugin?

**Comment by LeonDumestre** [comment]
Hi guys, I'm a novice with Ionic and I would like to use this plugin in my Ionic 6 app.
But when I added other plugins, I used to import the plugin with `/ngx` folder but I can't find it after ading and installing it.

How I could import it ? Did I do something wrong ?

Thanks 😄

**Reply by HarelM** [reply]
You don't need the ionic bindings for this plugin. I tried to open an issue with ionic native, but it was ignored.
I simply created the relevant bindings in javascript and added a typescript definition file to the npm package.
See here for example:
https://github.com/IsraelHikingMap/Site/blob/be13aa752b6da532eaa23be6b404a115e6de4009/IsraelHiking.Web/src/application/services/geo-location.service.ts#L12

**Reply by LeonDumestre** [reply]
Thanks a lot !

I'm sorry but I have another question, I added the plugin, I installed it and add the code as in your app:

```
import { BackgroundGeolocationPlugin } from "cordova-background-geolocation-plugin";
declare let BackgroundGeolocation: BackgroundGeolocationPlugin;
```

But when I run the app, the permissions are requested and it crashes on the first use of `BackgrounGeolocation` with this error:

> ERROR ReferenceError: BackgroundGeolocation is not defined
>     at AppComponent.initBackgroungGeolocation (main.js:129:9)
>     at new AppComponent (main.js:98:14)
>     at NodeInjectorFactory.AppComponent_Factory [as factory] (ɵfac.js? [sm]:1:1)
>     at getNodeInjectable (vendor.js:77426:44)
>     at instantiateRootComponent (vendor.js:84020:23)
>     at createRootComponent (vendor.js:86121:23)
>     at ComponentFactory.create (vendor.js:95459:25)
>     at ApplicationRef.bootstrap (vendor.js:100310:42)
>     at vendor.js:99993:64
>     at Array.forEach (<anonymous>)

Have you ever seen this error or do you have any idea what the problem is?

**Reply by HarelM** [reply]
Are you sure you are running this code in the `deviceready` event or afterwards?
In any case this seems like a bad compilation or bad import, not sure, I can assure you the code I linked is working in production in the play store and app store and works as expected.
BTW, feel free to send PRs to improve documentation or create and share a demo repo we can use to showcase this plugin.

**Reply by LeonDumestre** [reply]
It works!
I created a service to use it instead of doing it in app.component.

Thank you so much!

**Comment by Pazkov** [comment]
Hi all,
Google plays rejected my app few days ago because seems that my app should use only foreground geolocation. 
There is a way to completely disable background functionality?

thanks

**Reply by HarelM** [reply]
You can use the browser's geo location without this plugin or use it in foreground mode only, which is what I'm doing in my app...

**Reply by Pazkov** [reply]
Thanks HarelM.
Can you please better explain how to use this plugin in foreground mode only?

**Reply by HarelM** [reply]
This is how my code looks:
https://github.com/IsraelHikingMap/Site/blob/38575cc23edf3db111849d2ae38bcb52cf49d9f7/IsraelHiking.Web/src/application/services/geo-location.service.ts#L124
I don't use the background permissions.

**Reply by Pazkov** [reply]
Thanks a lot HarelM.
I’ll take a look on your code

**Comment by monbutiong** [comment]
Is this plugin incompatible with Gradle versions 8.0 and higher? I received the error message "Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0" when attempting to build the app with this plugin.

**Reply by HarelM** [reply]
Are you using the latest version? I think I was able to remove all these, didn't I?

**Comment by rosarioxp** [comment]
Hi everyone., My name is Rosario
I'm trying to use this plugin. It seems to work well in foreground, but crashes as soon as it enters the background mode. I'm testing on Android device.
Do you have any suggestions for solving this problem? Thank you

**Reply by HarelM** [reply]
Might be some permission issues, try and open Android studio and look in the logcat for the relevant exception.
I'm using this plugin in production, and it works well.
